### PR TITLE
Junit test for timestamps on French SQL Server

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
Related PR fixes issues with timestamps used in compare clauses on SQL Server. This adds a junit test for the fix.

#### Related Pull Requests
- https://github.com/LabKey/platform/pull/5987

#### Changes
- Optional ability to pass a `Connection` to a `TableSelector`
- Exercise timestamps with various compare clause scenarios. On SQL Server, exercise the test steps with connection set to English and then French.

### CODE REVIEW

File: bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java [hunk 1]
```diff
@@ -51,9 +67,16 @@ public void prepare(DbScope scope)
...
-        _dateFormat = (String) map.get("date_format");
+        try
+        {
+            LanguageSettings settings = getLanguageSettings(scope, scope.getConnection());
+            _language = settings.getLanguage();
+            _dateFormat = settings.getDate_format();
+        }
+        catch (SQLException e)
+        {
+            throw new RuntimeException(e);
```
Comment (labkey-jeckels) [inline bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java]: RuntimeSQLException?

**Review by labkey-jeckels — APPROVED** [review]

**Comment by labkey-tchad** [comment]
Verifying on [`Old Dependencies`](https://teamcity.labkey.org/buildConfiguration/LabKey_247Release_Internal_InternalSuites_OldDependenciesSqlserver/3245095?hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true) suite. If there were to be surprise failures, that seems like a likely place.

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
This seems to be causing a connection leak.

**Comment by labkey-tchad** [comment]
Yup, got a run with mem tracker stacktraces enabled:
```
org.labkey.api.util.MemTracker.put(MemTracker.java:371)
org.labkey.api.data.ConnectionWrapper.<init>(ConnectionWrapper.java:119)
org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase$1.getWrapped(MicrosoftSqlServer2016Dialect.java:365)
org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase$1.<init>(MicrosoftSqlServer2016Dialect.java:354)
org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase.testCompareClauses(MicrosoftSqlServer2016Dialect.java:352)
```

**Comment by labkey-adam** [comment]
> Yup, got a run with mem tracker stacktraces enabled:
> 
> ```
> org.labkey.api.util.MemTracker.put(MemTracker.java:371)
> org.labkey.api.data.ConnectionWrapper.<init>(ConnectionWrapper.java:119)
> org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase$1.getWrapped(MicrosoftSqlServer2016Dialect.java:365)
> org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase$1.<init>(MicrosoftSqlServer2016Dialect.java:354)
> org.labkey.bigiron.mssql.MicrosoftSqlServer2016Dialect$TestCase.testCompareClauses(MicrosoftSqlServer2016Dialect.java:352)
> ```

Yep, I see that. Will track it down...

**Comment by labkey-adam** [comment]
Note that the one BVT test run in red below passed in a subsequent run: https://teamcity.labkey.org/buildConfiguration/LabKey_247Release_Community_BvtBSqlserver/3251902?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false

**Review by labkey-tchad — APPROVED** [review]

File: bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java [hunk 2]
```diff
@@ -380,6 +365,36 @@ private Connection getWrapped() throws SQLException
                     testMultipleFilters(conn, containers, created.getFieldKey());
                 }
             }
+
+            // Null out connection to prevent query profiler from holding onto it via this scope
+            scope.clearConnection();
```
Comment (labkey-tchad) [inline bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java]: Maybe put this in a `try/finally` to make sure it gets cleared if the test fails.

**Review by labkey-tchad — APPROVED** [review]
This is making me very paranoid about anonymous classes.

File: bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java (hunk 2 above)
Comment (labkey-adam) [reply bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java]: @labkey-tchad The most recent version (what I merged earlier today) replaced the anonymous `DbScope` and `ConnectionWrapper` overrides with concrete subclasses. `TestScope` implements `AutoCloseable`, so the connection does get closed in a `try/finally`. Look it over and let me know if you have further feedback...

Comment (labkey-tchad) [reply bigiron/src/org/labkey/bigiron/mssql/MicrosoftSqlServer2016Dialect.java]: Oops. I was just looking at one commit instead of the entire PR.

**Pull request merged**